### PR TITLE
chore: move native-image.ts back into common

### PR DESCRIPTION
**Pull request by miniak**
#### Description of Change
It was split to separate browser and renderer implementation in #24802, which got undone in #30728 and we can merge them back into common.

#### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes

#### Release Notes
Notes: notes

### CODE REVIEW

**Review by zcbenz — APPROVED** [review]

**Review by codebytere — APPROVED** [review]

File: lib/common/api/module-list.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Common modules, please sort alphabetically
 export const commonModuleList: ElectronInternal.ModuleEntry[] = [
   { name: 'clipboard', loader: () => require('./clipboard') },
+  { name: 'nativeImage', loader: () => require('./native-image') },
```
Comment (Bogarde21) [inline lib/common/api/module-list.ts:4]: Duplicate of #

File: lib/sandboxed_renderer/api/module-list.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ export const moduleList: ElectronInternal.ModuleEntry[] = [
   },
   {
     name: 'nativeImage',
-    loader: () => require('@electron/internal/renderer/api/native-image')
+    loader: () => require('@electron/internal/common/api/native-image')
```
Comment (Bogarde21) [inline lib/sandboxed_renderer/api/module-list.ts:16]: [](url)

File: lib/common/api/module-list.ts (hunk 1 above)
Comment (Bogarde21) [inline lib/common/api/module-list.ts:4]: 1. 

- [ ] ``

File: lib/sandboxed_renderer/api/module-list.ts [hunk 3]
```diff
@@ -13,7 +13,7 @@ export const moduleList: ElectronInternal.ModuleEntry[] = [
   },
   {
     name: 'nativeImage',
```
Comment (Bogarde21) [inline lib/sandboxed_renderer/api/module-list.ts:15]: [](url)

Comment (Bogarde21) [reply lib/sandboxed_renderer/api/module-list.ts:15]: - [x] @Bogarde21

Comment (Bogarde21) [inline lib/sandboxed_renderer/api/module-list.ts:15]: [](url)

File: lib/renderer/api/native-image.ts [hunk 4]
```diff
@@ -1,3 +0,0 @@
-const { nativeImage } = process._linkedBinding('electron_common_native_image');
-
-export default nativeImage;
```
Comment (Bogarde21) [inline lib/renderer/api/native-image.ts:3]: [](url)

**Pull request merged**